python/services/github_service.py

```diff
@@ -65,6 +67,34 @@ def __init__(self, org_token: str, organisation_name: str) -> None:
         ), execute_timeout=60)
         self.organisation_name: str = organisation_name
 
+    def archive_inactive_repositories(self, repo_type_to_archive: str, last_active_cutoff_date: datetime,
+                                      allow_list: list[str]) -> None:
+        for repo in self.__get_repos_to_consider_for_archiving(repo_type_to_archive):
+            if self.__is_repo_ready_for_archiving(repo, last_active_cutoff_date, allow_list):
+                repo.edit(archived=True)
+
+    def __get_repos_to_consider_for_archiving(self, repository_type: str) -> list[Repository]:
+        repositories = list(
+            self.github_client_core_api.get_organization(self.organisation_name).get_repos(type=repository_type))
+        return [repository for repository in repositories if not (repository.archived or repository.fork)]
+
+    def __is_repo_ready_for_archiving(self, repository, last_active_cutoff_date, allow_list: list[str]) -> bool:
+        latest_commit_position = 0
+        commit = None
+        try:  # Try block needed as get_commits() can cause exception when a repository has no commits as GH returns negative result.
+            commit = repository.get_commits()[latest_commit_position]
+        except Exception:
+            logging.warning(
+                f"Manually check repository: {repository.name}. Reason: No commits in repository")
+            return False
+
+        if commit.author.date < last_active_cutoff_date:
+            if repository.name in allow_list:
+                logging.info(
+                    f"Skipping repository: {repository.name}. Reason: Present in allow list")
+                return False
+            return True
```

Comment:
I think you're missing a return statement at the end of this method. There's a path where `if commit.author.date < last_active_cutoff_date: == False` and nothing happens.

Reply:
Great spot! Yes, I'd missed a test case of `test_no_archive_when_repo_before_cutoff` 🧪

Technically, it still works since it returns `None`, which evaluates to false - which is what we want 😁

But let's not process our business estate on Python truthy/falsy rules 😂

I'll get the test in and a line to explicitly return `False` in that case 🚀

Reply:
Sorted 🧪🧪🧪🧪